src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/VisibleMemberTable.java

```diff
@@ -884,69 +879,40 @@ class LocalMemberTable {
                     case CONSTRUCTOR:
                         addMember(e, Kind.CONSTRUCTORS);
                         break;
-                    case ENUM_CONSTANT:
-                        addMember(e, Kind.ENUM_CONSTANTS);
-                        break;
                 }
             }
 
-            // Freeze the data structures
+            // protect from unintended change
             for (Kind kind : Kind.values()) {
-                orderedMembers.computeIfPresent(kind, (k, v) -> Collections.unmodifiableList(v));
-                orderedMembers.computeIfAbsent(kind, t -> List.of());
-
-                memberMap.computeIfPresent(kind, (k, v) -> Collections.unmodifiableMap(v));
-                memberMap.computeIfAbsent(kind, t -> Map.of());
+                orderedMembers.compute(kind, (k, v) -> v == null ? List.of() : Collections.unmodifiableList(v));
+                namedMembers.compute(kind, (k, v) -> v == null ? Map.of() : Collections.unmodifiableMap(v));
             }
         }
 
-        String getMemberKey(Element e) {
-            return new SimpleElementVisitor14<String, Void>() {
-                @Override
-                public String visitExecutable(ExecutableElement e, Void aVoid) {
-                    return e.getSimpleName() + ":" + e.getParameters().size();
```

Comment:
That was bizarre.

Reply:
Somewhat, yes. That computation complicated code more than it improved (some aspects of) it.